config/config.go

```diff
@@ -0,0 +1,140 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/jeremywohl/flatten"
+	"github.com/mcuadros/go-defaults"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/viper"
+)
+
+type Loader struct {
+	v *viper.Viper
+}
+
+type LoaderOption func(*Loader)
+
+func WithViper(in *viper.Viper) LoaderOption {
+	return func(l *Loader) {
+		l.v = in
+	}
+}
+
+func WithConfigName(in string) LoaderOption {
+	return func(l *Loader) {
+		l.v.SetConfigName(in)
+	}
+}
+
+func AddConfigPath(in string) LoaderOption {
```

Comment:
Should we use same prefixes? `With`

Reply:
This option can be used multiple times, so you can have multiple paths to check for config.

Reply:
Added a comment to function to make it more clear.
Or we can go with`WithAdditionalConfigPath`, wdyt?

Reply:
I think even using two or more WithConfigPath will also not be confusing.

Reply:
Alright then, renaming to `WithConfigPath`.